dali/pipeline/operator/operator.h

```diff
@@ -166,6 +168,26 @@ class DLL_PUBLIC OperatorBase {
 
   DISABLE_COPY_MOVE_ASSIGN(OperatorBase);
 
+
+  template<typename T>
+  T GetDiagnostic(const std::string &name) const {
+    return *any_cast<T *>(counters_.at(name));
+  }
+
+  template<typename T>
+  void RegisterDiagnostic(std::string name, T *val) {
+    using namespace std;  // NOLINT
+    static_assert(is_arithmetic_or_half<remove_reference_t<T>>::value || is_same<T, string>::value,
+                  "The eligible counter types are arithmetic types or string");
+    // Note the WAR for `any` bug - when T* is passed to
```

Comment:
Fix `any` instead - change the universal reference constructor to the following:
```c++
  template <typename T>
  any(T &&value) {  // NOLINT
    using U = std::remove_reference_t<T>;
    static_assert(std::is_copy_constructible<U>::value,
      "only copy-constructible types can be stored in 'any'");
    static_assert(std::is_destructible<U>::value,
      "objects stored in 'any' must be destructible");
    assign<U>(value);
  }
```

Reply:
Done